### PR TITLE
Add initial tutorial and Discourse landing pages

### PR DESCRIPTION
**Pull request by taurus-forever**
# Issue
Following Canonical best practices we need to prepare [Diátaxis](https://canonical.com/blog/diataxis-a-new-foundation-for-canonical-documentation) documentations. It is a first part here "Tutorial".

# Solution
Add initial documentation/tutorial as a part of the current sprint planning.

# Release Notes
Add tutorial to the MySQL operator.

### CODE REVIEW

File: docs/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+  status: completed
+  timing:
+    completed: 2023-01-29 22:56:16 +0000 UTC
+    enqueued: 2023-01-29 22:56:15 +0000 UTC
+    started: 2023-01-29 22:56:15 +0000 UTC
+```
+The root `password` should match whatever you passed in when you entered the command.
+
+## Integrations (Relations for Juju 2.9)
+Relations, or what Juju 3.0+ documentation [describes as Integration](https://juju.is/docs/sdk/integration), are the easiest way to create a user for MySQL in Charmed MySQL. Relations automatically create a username, password, and database for the desired user/application. As mentioned earlier in the [Access MySQL section](#access-mysql) it is a better practice to connect to MySQL via a specific user rather than the admin user.
```
Comment (shayancanonical) [inline docs/tutorial.md:342]: describes as Integration -> describes as an Integration

**Review by shayancanonical — APPROVED** [review]
LGTM, just some minor nits

File: docs/documentation_landing.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+MySQL is the world’s most popular open source database. A relational database stores data in separate tables rather than putting all the data in one big storeroom. The database structure is organized into physical files optimized for speed. The logical data model, with objects such as data tables, views, rows, and columns, offers a flexible programming environment.
+
+This MySQL operator charm comes in two flavours to deploy and operate MySQL on [physical/virtual machines](https://github.com/canonical/mysql-operator) and [Kubernetes](https://github.com/canonical/mysql-k8s-operator). Both offer features such as replication, TLS, password rotation, and easy to use integration with applications. The Charmed MySQL Operator meets the need of deploying MySQL in a structured and consistent manner while allowing the user flexibility in configuration. It simplifies deployment, scaling, configuration and management of MySQL in production at scale in a reliable way.
+
+[note type="positive"]
+**"Charmed MySQL" or "Mysql Innodb Cluster"?**
```
Comment (paulomach) [inline docs/documentation_landing.md:9]: iirc the official stance is all things "charmed"

File: docs/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+App    Version          Status  Scale  Charm  Channel  Rev  Exposed  Message
+mysql  8.0.32-0ubun...  active      1  mysql  edge      95  no       Unit is ready: Mode: RW
+
+Unit      Workload  Agent  Machine  Public address  Ports  Message
+mysql/0*  active    idle   0        10.234.188.135         Unit is ready: Mode: RW
+
+Machine  State    Address         Inst id        Series  AZ  Message
+0        started  10.234.188.135  juju-ff9064-0  jammy       Running
+```
+To exit the screen with `juju status --watch 1s`, enter `Ctrl+c`.
```
Comment (paulomach) [inline docs/tutorial.md:104]:
```suggestion
To exit the screen with `juju status --watch 1s`, enter `Ctrl+c`.
If you want to further inspect juju logs, can watch for logs with `juju debug-log`. More info on logging at [juju logs](https://juju.is/docs/olm/juju-logs) 
```

File: docs/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+
+## Access MySQL
+> **!** *Disclaimer: this part of the tutorial accesses MySQL via the `root` user. **Do not** directly interface with the root user in a production environment. In a production environment always create a separate user using [Data Integrator](https://charmhub.io/data-integrator) and connect to MySQL with that user instead. Later in the section covering Relations we will cover how to access MySQL without the root user.*
+
+The first action most users take after installing MySQL is accessing MySQL. The easiest way to do this is via the [MySQL Command-Line Client](https://dev.mysql.com/doc/refman/8.0/en/mysql.html) `mysql`. Connecting to the database requires that you know the values for `host`, `username` and `password`. To retrieve the necessary fields please run Charmed MySQL action `get-password`:
+```shell
+juju run-action mysql/leader get-password --wait
```
Comment (paulomach) [inline docs/tutorial.md:111]: We are defaulting to root, but how about to keep it explicit?
```suggestion
juju run-action mysql/leader get-password username=root --wait
```

File: docs/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+juju relate data-integrator mysql
+```
+Re-relating generates a new user and password:
+```shell
+juju run-action data-integrator/leader get-credentials --wait
+```
+You can connect to the database with this new credentials.
+From here you will see all of your data is still present in the database.
+
+## Transcript Layer Security (TLS)
```
Comment (paulomach) [inline docs/tutorial.md:480]:
```suggestion
## Transport Layer Security (TLS)
```

**Review by paulomach — APPROVED** [review]
lgtm, minor comments/fixes

File: docs/documentation_landing.md (hunk 2 above)
Comment (taurus-forever) [reply docs/documentation_landing.md:9]: The idea here is to describe list already available MySQL/MariaDB charms explaining "Charmed MySQL" replaces all of them.

File: docs/tutorial.md (hunk 3 above)
Comment (taurus-forever) [reply docs/tutorial.md:104]: Applied.

File: docs/tutorial.md (hunk 5 above)
Comment (taurus-forever) [reply docs/tutorial.md:480]: @MiaAltieri , we need to fix this in [MongoDB](https://github.com/canonical/mongodb-operator/blob/main/documentation/tutorial.md#transcript-layer-security-tls) operator too... 

Fixed for MySQL.

File: docs/tutorial.md (hunk 4 above)
Comment (taurus-forever) [reply docs/tutorial.md:111]: Good point, TIL. Applied.

File: docs/documentation_landing.md (hunk 2 above)
Comment (taurus-forever) [reply docs/documentation_landing.md]: Please check the new list of charms mentioned there, I have tried to make it better readable.

**Review by paulomach — APPROVED** [review]

**Pull request merged**